hedging/src/rebalance_action.rs

```diff
@@ -0,0 +1,763 @@
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum RebalanceAction {
+    DoNothing,
+    TransferTradingToFunding(Decimal),
+    TransferFundingToTrading(Decimal),
+    OnchainDeposit(Decimal),
+    OnchainWithdraw(Decimal),
+}
+impl std::fmt::Display for RebalanceAction {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RebalanceAction::DoNothing => write!(f, "DoNothing"),
+            RebalanceAction::TransferTradingToFunding(amount_in_btc) => {
+                write!(f, "TransferTradingToFunding({})", amount_in_btc)
+            }
+            RebalanceAction::TransferFundingToTrading(amount_in_btc) => {
+                write!(f, "TransferFundingToTrading({})", amount_in_btc)
+            }
+            RebalanceAction::OnchainDeposit(amount_in_btc) => {
+                write!(f, "OnchainDeposit({})", amount_in_btc)
+            }
+            RebalanceAction::OnchainWithdraw(amount_in_btc) => {
+                write!(f, "OnchainWithdraw({})", amount_in_btc)
+            }
+        }
+    }
+}
+impl RebalanceAction {
+    pub fn action_required(&self) -> bool {
+        !matches!(*self, Self::DoNothing)
+    }
+
+    pub fn action_type(&self) -> &'static str {
+        match *self {
+            Self::DoNothing => "do-nothing",
+            Self::TransferTradingToFunding(_) => "transfer-trading-to-funding",
+            Self::TransferFundingToTrading(_) => "transfer-funding-to-trading",
+            Self::OnchainDeposit(_) => "deposit",
+            Self::OnchainWithdraw(_) => "withdraw",
+        }
+    }
+
+    pub fn size(&self) -> Option<Decimal> {
+        match *self {
+            Self::TransferTradingToFunding(size)
+            | Self::TransferFundingToTrading(size)
+            | Self::OnchainDeposit(size)
+            | Self::OnchainWithdraw(size) => Some(size),
+            _ => None,
+        }
+    }
+
+    pub fn unit(&self) -> &'static str {
+        "btc"
+    }
+}
+
+fn round_btc(amount_in_btc: Decimal) -> Decimal {
+    let amount_in_sats = amount_in_btc * SATS_PER_BTC;
+    amount_in_sats.round() / SATS_PER_BTC
+}
+
+fn floor_btc(amount_in_btc: Decimal) -> Decimal {
+    let amount_in_sats = amount_in_btc * SATS_PER_BTC;
+    amount_in_sats.floor() / SATS_PER_BTC
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn determine_action(
+    abs_liability_in_cents: SyntheticCentLiability,
+    signed_exposure_in_cents: SyntheticCentExposure,
+    total_collateral_in_btc: Decimal,
+    btc_price_in_cents: Decimal,
+    funding_btc_total_balance: Decimal,
+    internal_transfer_pending: bool,
+    external_transfer_pending: bool,
+    HedgingFundingConfig(hedging_config, funding_config): HedgingFundingConfig,
+) -> RebalanceAction {
+    let abs_liability_in_btc = abs_liability_in_cents / btc_price_in_cents;
+    let abs_exposure_in_btc = Decimal::from(signed_exposure_in_cents).abs() / btc_price_in_cents;
+    if abs_exposure_in_btc.is_zero()
```

Comment:
I think this whole block of `if .. else if ... else if` etc. can probably be cleaned up via 1 level of nesting or a `match` clause or perhaps sub functions that describe what the branch signifies. As is its not possible for me to understand what condition is actually being checked here.

Reply:
I have factored out the second level into sub functions finally, 
each first level conditions have a specific  meaning that would be lost otherwise. 
But each branch is either a deposit or a withdrawal action (and "internal"/"external" depends on the funding available)